server.go

```diff
@@ -301,6 +304,9 @@ type Server struct {
 
 	inFlight sync.WaitGroup
 
+	shutdownSignal chan os.Signal
```

Comment:
This is just a boolean that you're sending?

Reply:
Defined as a boolean in the previous version, but consider to further use, so I use Signal instead. Actually boolean is better in current usage.

Reply:
Just minimize the PR's size, but not doing this change now (if at all)

Reply:
yep, try to remove this variable

Reply:
I tried but failed. I think we should add a chan bool for shutdown signal finally, also simplify the flow of every property. what do u think